### PR TITLE
Enhance entry point style docs

### LINKED ISSUE #349: Enhance documentation about entry point styles

### PR DESCRIPTION
**Pull request by chrisjs**
Resolves #349

### CODE REVIEW

File: spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc [hunk 1]
```diff
@@ -1181,9 +1181,11 @@ Replace `<app>` with the name of your application and the value of your annotati
 
 An Entry Point Style affects how application properties are passed to the container to be deployed. Currently, three styles are supported:
 
-* `exec`: (default) Passes all application properties as command line arguments.
-* `shell`: Passes all application properties as environment variables.
-* `boot`: Creates an environment variable called `SPRING_APPLICATION_JSON` that contains a JSON representation of all application properties.
+* `exec`: (default) Passes all application properties and command line arguments in the deployment request as container args. Application properties are transformed into the format of `--key=value`.
+* `shell`: Passes all application properties as environment variables. Command line arguments from the deployment request are not converted into environment variables nor set on the container. Application properties are transformed into an uppercase string and `.` characters are replaced with `_`.
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc:1185]: > Command line arguments from the deployment request are not converted into environment variables nor set on the container.

maybe:

> Command line arguments from the deployment request are not converted into environment variables nor set on the container - they are skipped.

File: spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc [hunk 2]
```diff
@@ -1181,9 +1181,11 @@ Replace `<app>` with the name of your application and the value of your annotati
...
+* `boot`: Creates an environment variable called `SPRING_APPLICATION_JSON` that contains a JSON representation of all application properties. Command line arguments from the deployment request are set as container args.
+
+NOTE: In all cases, environment variables defined at the server level configuration and on a per-application basis are set onto the container as-is.
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc:1188]: Perhaps, a missing "are" in between "variables defined".

File: spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc (hunk 1 above)
Comment (chrisjs) [reply spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc:1185]: theres no skipping logic or anything, theres simply nothing done with them so id suggest not stating 'they are skipped'

File: spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc (hunk 2 above)
Comment (chrisjs) [reply spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc:1188]: ` In all cases, environment variables defined at the server level` is stating one of the places env vars come from not how to define them

**Pull request merged**